src/socketio.quiver.js

```diff
@@ -144,11 +144,48 @@ QuiverSocialProvider.configuration_ = undefined;
  */
 QuiverSocialProvider.connection_ = undefined;
 
+/**
+ * @param {!Array<T>} list To sample from
+ * @param {number} k Target number of samples
+ * @return {!Array<T>} Subset of |list|
+ * @template T
+ */
+QuiverSocialProvider.randomSample_ = function(list, k) {
+  if (k >= list.length) {
+    return list;
+  }
+
+  // Algorithm R: https://en.wikipedia.org/wiki/Reservoir_sampling#Algorithm_R
+  var sample = list.slice(0, k);
+  for (var i = k; i < list.length; ++i) {
+    var j = Math.floor(Math.random() * i);
```

Comment:
I'm struggling to wrap my brain around this algorithm, but I'm seeing that if this algorithm is called with a list.length==7 and k==2, it is biased against the first 2 elements of the list (the latter 5 elements have about 2x chance of getting sampled than the first 2 elements).  I have tests that prove this - I can demo